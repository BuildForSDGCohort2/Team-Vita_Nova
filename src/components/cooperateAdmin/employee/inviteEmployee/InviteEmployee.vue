<template>
  <v-card class="mx-auto mt-4" width="500" rounded>
    <validationObserver>
      <v-form>
        <v-card-title class="text-center justify-center">
          <h3>Invite Employees</h3>
        </v-card-title>
        <v-file-input
          v-model="files"
          accept=".csv, .xlsx"
          class="pt-8 px-6"
          label="Upload CSV Document"
          multiple
          filled
          prepend-inner-icon="mdi-cloud-upload"
          prepend-icon
        >
          <template v-slot:selection="{ text }">
            <v-chip small label color="primary">{{ text }}</v-chip>
          </template>
        </v-file-input>

        <h3 class="text-center pt-8">OR</h3>

        <p class="text-center pt-8">Manually input Employees Email</p>
        <ValidationProvider rules="required|email">
          <v-combobox
            class="px-6"
            v-model="model"
            :filter="filter"
            :hide-no-data="!search"
            :search-input.sync="search"
            hide-selected
            label="Email Address"
            multiple
            small-chips
            solo
            :loading="load"
          >
            <template v-slot:no-data>
              <v-list-item>
                <span class="subheading">Create</span>
                <v-chip :color="`red lighten-3`" label small>{{
                  search
                }}</v-chip>
              </v-list-item>
            </template>
            <template v-slot:selection="{ attrs, item, parent, selected }">
              <v-chip
                v-if="item === Object(item)"
                v-bind="attrs"
                :color="`red lighten-3`"
                :input-value="selected"
                label
                small
              >
                <span class="pr-2">{{ item.text }}</span>
                <v-icon small @click="parent.selectItem(item)"
                  >mdi-close</v-icon
                >
              </v-chip>
            </template>
            <template v-slot:item="{ index, item }">
              <v-text-field
                v-if="editing === item"
                v-model="editing.text"
                height="60"
                autofocus
                flat
                background-color="transparent"
                hide-details
                solo
                @keyup.enter="edit(index, item)"
              ></v-text-field>
              <v-chip
                v-else
                :color="`${item.color} lighten-3`"
                dark
                label
                small
                >{{ item.text }}</v-chip
              >
              <v-spacer></v-spacer>
              <v-list-item-action @click.stop>
                <v-btn icon @click.stop.prevent="edit(index, item)">
                  <v-icon>{{
                    editing !== item ? "mdi-pencil" : "mdi-check"
                  }}</v-icon>
                </v-btn>
              </v-list-item-action>
            </template>
          </v-combobox>
        </ValidationProvider>
      </v-form>

      <v-btn
        color="#FF2E2E"
        dark
        width="400"
        height="60"
        class="my-8 ml-12"
        @click.prevent="inviteEmployee"
        >Send Invite(s)</v-btn
      >
    </validationObserver>
  </v-card>
</template>

<script>
import UserService from "../../../../services/user-services";
export default {
  data: () => ({
    files: [],
    activator: null,
    value: "",
    attach: null,
    colors: ["green", "purple", "indigo", "cyan", "teal", "orange"],
    editing: null,
    index: -1,
    load: false,
    items: [
      { header: "Select an option or create one" },
      {
        text: "Foo",
        color: "blue"
      },
      {
        text: "Bar",
        color: "red"
      }
    ],
    nonce: 1,
    menu: false,
    model: [],
    x: 0,
    search: null,
    y: 0
  }),

  watch: {
    model(val, prev) {
      if (val.length === prev.length) return;

      this.model = val.map(v => {
        if (typeof v === "string") {
          v = {
            text: v,
            color: this.colors[this.nonce - 1]
          };

          this.items.push(v);

          this.nonce++;
        }

        return v;
      });
    }
  },

  methods: {
    edit(index, item) {
      if (!this.editing) {
        this.editing = item;
        this.index = index;
      } else {
        this.editing = null;
        this.index = -1;
      }
    },
    filter(item, queryText, itemText) {
      if (item.header) return false;

      const hasValue = val => (val != null ? val : "");

      const text = hasValue(itemText);
      const query = hasValue(queryText);

      return (
        text
          .toString()
          .toLowerCase()
          .indexOf(query.toString().toLowerCase()) > -1
      );
    },
    inviteEmployee() {
      this.load = true;
      const data = { emails: this.model };

      UserService.inviteEmployee(data, this.files)
        .then(res => {
          console.log(res.data);
          this.load = false;
          this.$emit("showSplash");
        })
        .catch(err => {
          console.log(err);
        });
    }
  }
};
</script>
