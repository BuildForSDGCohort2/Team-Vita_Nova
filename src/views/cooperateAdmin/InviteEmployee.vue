<template>
  <InviteEmployeeForm />
</template>

<script>
import InviteEmployeeForm from "../../components/cooperateAdmin/employee/inviteEmployee/InviteEmployeeForm";
export default {
  components: {
    InviteEmployeeForm
  }
};
</script>