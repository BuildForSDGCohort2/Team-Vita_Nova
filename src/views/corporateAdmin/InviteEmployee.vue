<template>
  <InviteEmployeeForm />
</template>

<script>
import InviteEmployeeForm from "../../components/corporateAdmin/employee/inviteEmployee/InviteEmployeeForm";
export default {
  components: {
    InviteEmployeeForm
  }
};
</script>