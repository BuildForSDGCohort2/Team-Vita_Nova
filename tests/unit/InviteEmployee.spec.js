import InviteEmployee from "../../src/components/cooperateAdmin/employee/inviteEmployee/InviteEmployeeForm"
import { mount} from '@vue/test-utils'

describe('InviteEmployee',()=>{
    test('setup correctly', () =>{
        expect(true).toBe(true)
    })
})

describe('InviteEmployee', ()=>{
    const wrapper=mount(InviteEmployee)
    it('html should render correctly', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    //test page title
    it('test page has title', () =>{
        expect(wrapper.text()).toContain('Invite your Employees')
    })

    // Inspect the raw component options
    it("has a created hook", () => {
        expect(typeof InviteEmployee.mounted).toBe("function");
    });

    it("sets the correct default data", () => {
        expect(typeof InviteEmployee.data).toBe("function");
        const defaultData = InviteEmployee.data();
        expect(defaultData.email).toBe("");
    });
    it("renders a vue instance", () => {
        expect(mount(InviteEmployee).isVueInstance()).toBe(true);
    });
    it("Checks the data-title", () => {
        expect(wrapper.vm.title).toMatch("inviteEmployee");
    });

})