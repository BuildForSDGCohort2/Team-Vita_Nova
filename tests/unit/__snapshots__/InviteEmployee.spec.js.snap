// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InviteEmployee html should render correctly 1`] = `
"<v-row justify=\\"center\\">
  <v-dialog persistent=\\"\\" max-width=\\"600px\\">
    <v-card class=\\"invite-card ma-auto\\">
      <v-row>
        <v-col cols=\\"12\\" md=\\"2\\">
          <v-btn icon=\\"\\" class=\\"mx-0\\" style=\\"float: right;\\">
            <v-icon>mdi-close-circle-outline</v-icon>
          </v-btn>
        </v-col>
        <v-col cols=\\"12\\" md=\\"10\\">
          <v-row>
            <v-card-title class=\\"invite-title\\">
              <h3>Invite your Employees</h3>
            </v-card-title>
          </v-row>
        </v-col>
      </v-row>
      <v-card-text>
        <validationobserver></validationobserver>
      </v-card-text>
    </v-card>
  </v-dialog>
</v-row>"
`;
